server/src/test/java/org/elasticsearch/cluster/routing/allocation/ShardsAvailabilityHealthIndicatorServiceTests.java

```diff
@@ -1944,6 +2036,30 @@ private static IndexRoutingTable index(String name, ShardAllocation primaryState
         );
     }
 
+    private static IndexRoutingTable index(
```

Comment:
Nit: Should we rename this method instead of overloading `index(String, ShardAllocation, ShardAllocation...)`.
I have to admit, that if later in time I would be working on these test, I might miss/forget that when I call it with 4 ShardAllocation it behaves differently than when I call it with less or more.

- `index("myindex", new ShardAllocation(...) x1)` -> 1 primary shard
- `index("myindex", new ShardAllocation(...) x2)` -> 1 primary shard & 1 replica shard
- `index("myindex", new ShardAllocation(...) x3)` -> 1 primary shard & 2 replica shards
- `index("myindex", new ShardAllocation(...) x4)` -> 2 primary shard & 2 replica shard
- `index("myindex", new ShardAllocation(...) x5)` -> 1 primary shard & 4 replica shard

What do you think?

Or maybe we could merge these two methods to something else:
`index(String name, Tuple<ShardAllocation primaryState, List<ShardAllocation>>, Tuple<ShardAllocation primaryState, List<ShardAllocation>>...)`, this way we group the primary shard with its replicas, if any.

Reply:
I'll rename this to (hopefully) make it clearer, thanks!